apps/prairielearn/elements/pl-excalidraw/pl-excalidraw.py

```diff
@@ -0,0 +1,100 @@
+import base64
+import json
+from pathlib import Path
+
+import chevron
+import lxml.html
+import prairielearn as pl
+
+
+def prepare(element_html: str, _: pl.QuestionData) -> None:
+    element = lxml.html.fragment_fromstring(element_html)
+    pl.check_attribs(element, ["name"], ["width", "height", "file", "file_dir"])
```

Comment:
Conventionally I think we'd update the attributes as follows:

- `name` -> `answers-name`
- `file` -> `source-file-name`
- `file_dir` -> `directory`

This is based on other elements, which you can see in `docs/elements.md`.

We'd also typically break out the second/third arguments into local variables for clarity, e.g.:

https://github.com/PrairieLearn/PrairieLearn/blob/9dda505756eded11fde4a69561eeebdc915b1aee/apps/prairielearn/elements/pl-checkbox/pl-checkbox.py#L27-L47

You'll also want a call to `pl.check_answers_names`, e.g.:

https://github.com/PrairieLearn/PrairieLearn/blob/9dda505756eded11fde4a69561eeebdc915b1aee/apps/prairielearn/elements/pl-checkbox/pl-checkbox.py#L48-L49

cc @eliotwrobson if you want to chime in on element "standardization" stuff.

Reply:
Big +1 on trying to keep the names / behavior consistent with that of other elements. I'll refer back to this in some of my other comments, but a good example to work from is the `pl-figure` element, since it uses some similar attributes to this element: https://github.com/PrairieLearn/PrairieLearn/blob/master/apps/prairielearn/elements/pl-figure/pl-figure.py

Reply:
`pl.check_answers_names(data, name)` does not work with my existing question template - it is an overapproximation and does not distinguish duplicate names across disjoint panel types.

```html
<pl-question-panel>
 <pl-excalidraw answers-name="vector"></pl-excalidraw>
</pl-question-panel>

<pl-answer-panel>
 <pl-excalidraw answers-name="vector"></pl-excalidraw>
</pl-answer-panel>
```

Here, I want both elements to co-exist. One might ask why I can't hoist the `pl-excalidraw` outside the panel types if I am duplicating it anyway - if I need different styles / layouts around the same figure in different panels, I should be allowed to do so - maybe I want a different width in the answer panel. I know it's semantically equivalent to using `pl-hide-in-panel`, so I'll switch to it.

Reply:
Yep, `pl-hide-in-panel` is the way to go here, and is exactly what we'd do for other elements that use `pl.check_answers_names` to enforce uniqueness.